woocommerce/woocommerce
# Fixed the case conversion for meta key column mapping in product CSV imports

### All Submissions:






### Changes proposed in this Pull Request:

This is another pass at #25517 

This PR leaves the case insensitivity for column names in place, while also solving the case conversion that was occurring in special columns. Since the regular expression was being ran against the normalized field, it was converting meta keys and the like to lowercase as well. We can solve this by using a case-insensitive regex instead, and running it against the original column input.

### How to test the changes in this Pull Request:

1. Make a change to the `sample-products.csv` file's `Meta: _wpcom_is_markdown` header. Change the casing to something like `Meta: _WPCOM_IS_MARKDOWN` and save.
2. Import this file and check the database for the meta key. Without the PR, note that it has been converted to lowercase. With the PR, notice that it is intact.
3. Make sure all of the products are imported as expected.

### Other information:




### Changelog entry

> Fix - Removed case conversion of meta keys from CSV imports.
